Mathlib/NumberTheory/LSeries/HurwitzZetaOdd.lean

```diff
@@ -0,0 +1,592 @@
+    continuous_const.mul continuous_ofReal⟩).continuousAt
+
+lemma oddKernel_functional_equation (a : UnitAddCircle) (x : ℝ) :
+    oddKernel a x = 1 / x ^ (3 / 2 : ℝ) * sinKernel a (1 / x) := by
+  -- first reduce to `0 < x`
+  rcases le_or_lt x 0 with hx | hx
+  · rw [oddKernel_undef _ hx, sinKernel_undef _ (one_div_nonpos.mpr hx), mul_zero]
+  induction' a using QuotientAddGroup.induction_on' with a
+  have h1 : -1 / (I * ↑(1 / x)) = I * x := by rw [one_div, ofReal_inv, mul_comm, ← div_div,
+    div_inv_eq_mul, div_eq_mul_inv, inv_I, mul_neg, neg_one_mul, neg_mul, neg_neg, mul_comm]
+  have h2 : (-I * (I * ↑(1 / x))) = 1 / x := by
+    rw [← mul_assoc, neg_mul, I_mul_I, neg_neg, one_mul, ofReal_div, ofReal_one]
+  have h3 : (x : ℂ) ^ (3 / 2 : ℂ) ≠ 0 := by
+    simp only [Ne, cpow_eq_zero_iff, ofReal_eq_zero, hx.ne', false_and, not_false_eq_true]
+  have h4 : arg x ≠ π := by rw [arg_ofReal_of_nonneg hx.le]; exact pi_ne_zero.symm
+  rw [← ofReal_inj, oddKernel_def, ofReal_mul, sinKernel_def, jacobiTheta₂'_functional_equation',
+    h1, h2]
+  generalize jacobiTheta₂'' a (I * ↑x) = J
+  rw [one_div (x : ℂ), inv_cpow _ _ h4, div_inv_eq_mul, one_div, ofReal_inv, ofReal_cpow hx.le,
+    ofReal_div, ofReal_ofNat, ofReal_ofNat, ← mul_div_assoc _ _ (-2 * π : ℂ),
+    eq_div_iff <| mul_ne_zero (neg_ne_zero.mpr two_ne_zero) (ofReal_ne_zero.mpr pi_ne_zero),
+    ← div_eq_inv_mul, eq_div_iff h3, mul_comm J _, mul_assoc (-2 * π : ℂ) ((x : ℂ) ^ _),
+    mul_comm ((x : ℂ) ^ _) J, ← mul_assoc]
+
+end kernel_defs
+
+section sum_formulas
+
+/-!
+## Formulae for the kernels as sums
+-/
+
+lemma hasSum_int_oddKernel (a : ℝ) {x : ℝ} (hx : 0 < x) :
+    HasSum (fun n : ℤ ↦ (n + a) * rexp (-π * (n + a) ^ 2 * x)) (oddKernel ↑a x) := by
+  rw [← hasSum_ofReal, oddKernel_def' a x]
+  have h1 := hasSum_jacobiTheta₂_term (a * I * x) (by rwa [I_mul_im, ofReal_re] : 0 < im (I * x))
+  have h2 := hasSum_jacobiTheta₂'_term (a * I * x) (by rwa [I_mul_im, ofReal_re] : 0 < im (I * x))
+  refine (((h2.div_const (2 * π * I)).add (h1.mul_left ↑a)).mul_left
+    (cexp (-π * a ^ 2 * x))).congr_fun (fun n ↦ ?_)
+  rw [jacobiTheta₂'_term, mul_assoc (2 * π * I), mul_div_cancel_left₀ _ two_pi_I_ne_zero, ← add_mul,
+    ← mul_assoc, mul_comm _ (n + a : ℂ), mul_assoc (n + a : ℂ), jacobiTheta₂_term,
+    ← Complex.exp_add]
+  push_cast
+  congr 2
+  simp only [← mul_assoc, ← add_mul]
+  congr 1
+  simp_rw [mul_assoc _ I, mul_comm I _, ← mul_assoc _ _ I, mul_assoc _ I, mul_comm I _,
+    ← mul_assoc _ _ I, add_mul, mul_assoc _ _ I, I_mul_I, mul_neg, mul_one,
+    mul_comm (2 : ℂ) π, mul_assoc, ← neg_mul, ← mul_add]
+  rw [← add_assoc, add_comm, mul_comm (n : ℂ) a, ← mul_assoc, add_sq]
+
+lemma hasSum_int_sinKernel (a : ℝ) {t : ℝ} (ht : 0 < t) : HasSum
+    (fun n : ℤ ↦ -I * n * cexp (2 * π * I * a * n) * rexp (-π * n ^ 2 * t)) ↑(sinKernel a t) := by
+  have h : -2 * (π : ℂ) ≠ (0 : ℂ) := by
+    simp? [pi_ne_zero] says simp only [neg_mul, ne_eq, neg_eq_zero, mul_eq_zero,
```

Comment:
Did you use the `says` combinator here on purpose? Just asking, because it is the only place where you do that.

Reply:
I meant to take it out after doing a speed-test.